SparseGrids/gridtestExternalTests.cpp

```diff
@@ -215,101 +262,125 @@ TestResults ExternalTester::getError(const BaseFunction *f, TasGrid::TasmanianSp
             err += std::abs(y[j] - r[j]);
         };
         R.error = err;
-    }else if (type == type_internal_interpolation){
-        // load needed points
-        loadValues(f, grid);
-
-        std::vector<double> err(num_outputs, 0.0); // absolute error
-        std::vector<double> nrm(num_outputs, 0.0); // norm, needed to compute relative error
-
-        std::vector<double> test_x = genRandom(num_mc, num_dimensions);
-        std::vector<double> result_tasm(num_mc * num_outputs);
-        std::vector<double> result_true(num_mc * num_outputs);
+    }else if (type == type_internal_interpolation or type == type_internal_differentiation){
+        if (type == type_internal_differentiation and (!grid.isGlobal() or grid.isSequence())) {
+            // Avoid testing grids where derivatives have not been implemented.
+            R.error = 0.0;
+        } else {
+            // load needed points
+            loadValues(f, grid);
+
+            std::vector<double> test_x = genRandom(num_mc, num_dimensions);
+            int num_entries = (type == type_internal_interpolation) ? num_outputs : num_outputs * num_dimensions;
+            std::vector<double> result_tasm(num_mc * num_entries);
+            std::vector<double> result_true(num_mc * num_entries);
+            std::vector<double> err(num_entries, 0.0); // absolute error
```

Comment:
Similar story to above, no need to store the errors and norms into a vector. The only time when this is needed is when you are inside a parallel region and threads need to write to different locations.

Reply:
Changed. Note that this logic for the error vectors existed before this PR.